applications/matter_bridge/src/ble_providers/ble_bridged_device_factory.cpp

```diff
@@ -334,10 +335,14 @@ CHIP_ERROR BleBridgedDeviceFactory::CreateDevice(int deviceType, bt_addr_le_t bt
 	BLEConnectivityManager::Instance().Recover(provider);
 	err = AddMatterDevices(reinterpret_cast<MatterBridgedDevice::DeviceType *>(&deviceType), 1, nodeLabel, provider,
 			       &index, &endpointId);
-exit:
 
 	if (err != CHIP_NO_ERROR) {
 		BLEConnectivityManager::Instance().RemoveBLEProvider(btAddress);
```

Comment:
Ditto

Reply:
It seems like we have one more `BLEConnectivityManager::Instance().RemoveBLEProvider(btAddress);` here.

Reply:
Yes, but this one is a bit more tricky. I was not sure if we really end up with discovery callbacks being triggered in the recovery path unconditionally.

Reply:
Discussed f2f.